wrappers/golang/curves/bls12377/vec_ops_test.go

```diff
@@ -31,3 +32,37 @@ func TestVecOps(t *testing.T) {
 
 	assert.Equal(t, a, out3)
 }
+
+func TestTranspose(t *testing.T) {
+
+	rowSize := 1 << 6
+	columnSize := 1 << 8
+	onDevice := false
+	isAsync := true
+
+	matrix := GenerateScalars(rowSize * columnSize)
+
+	out := make(core.HostSlice[ScalarField], rowSize*columnSize)
+	out2 := make(core.HostSlice[ScalarField], rowSize*columnSize)
+
+	ctx, _ := cr.GetDefaultDeviceContext()
+
+	TransposeMatrix(matrix, out, columnSize, rowSize, ctx, onDevice, isAsync)
+	TransposeMatrix(out, out2, rowSize, columnSize, ctx, onDevice, isAsync)
+
+	assert.Equal(t, matrix, out2)
+
+	var d_matrix, d_out, d_out2 core.DeviceSlice
+	onDevice = true
+
+	matrix.CopyToDeviceAsync(&d_matrix, *ctx.Stream, true)
```

Comment:
We should create a new stream since the default stream is the "main" thread on the GPU.